File: dotnet/tests/amp.tests/rabbit/topology/ConsumingRouteTests.cs
using amp.rabbit.topology;
using NUnit.Framework;

namespace amp.tests.rabbit.topology
{
    /// <summary>
    /// These tests assert that equality and hashcode generation use value semantics
    /// not just in-so-far as properties have the same reference values but that 
    /// properties themselves are compared using a value semantic.  
    /// </summary>
    public class ConsumingRouteTests : BaseRouteTests
    {
        protected ConsumingRoute _queueNotEquivelentModel;

        public override void SetUp()
        {
            _referenceModel = NewConsummingRoute("broker1", "exchange1", "queue1", "route1");
            _equivelentModel = NewConsummingRoute("broker1", "exchange1", "queue1", "route1");
            _brokerNotEquivelentModel = NewConsummingRoute("broker2", "exchange1", "queue1", "route1");
            _exchangeNotEquivelentModel = NewConsummingRoute("broker1", "exchange2", "queue1", "route1");
            _routeNotEquivelentModel = NewConsummingRoute("broker1", "exchange1", "queue1", "route2");
            _queueNotEquivelentModel = NewConsummingRoute("broker1", "exchange1", "queue2", "route1");
        }

        private ConsumingRoute NewConsummingRoute(string host, string exchange, string queue, string routeKey)
        {
            return new ConsumingRoute(new[] { new Broker(host, 0) }, new Exchange(exchange), new Queue(queue), new[] { routeKey });
        }


        [Test]
        public void Equals_should_return_false_if_queue_values_differ()
        {
            Assert.That(_queueNotEquivelentModel.Equals(_referenceModel), Is.False);
        }

        [Test]
        public void Hashcodes_should_differ_if_queue_entries_differ()
        {
            Assert.That(_queueNotEquivelentModel.GetHashCode(), Is.Not.EqualTo(_equivelentModel.GetHashCode()));
        }

    }
}

File: java/amp-rabbit/src/test/java/amp/rabbit/topology/ConsumingRouteTests.java
package amp.rabbit.topology;

import static org.junit.Assert.*;

import org.junit.*;

/** 
*  These tests assert that equality and hashcode generation use value semantics
*  not just in-so-far as properties have the same reference values but that 
* properties themselves are compared using a value semantic.  
*/
public class ConsumingRouteTests extends BaseRouteTests {
	
	protected ConsumingRoute _queueNotEquivelentModel;

	@Override 
	public void setup() {
		
        _referenceModel = newConsummingRoute("broker1", "exchange1", "queue1", "route1");
        _equivelentModel = newConsummingRoute("broker1", "exchange1", "queue1", "route1");
        _brokerNotEquivelentModel = newConsummingRoute("broker2", "exchange1", "queue1", "route1");
        _exchangeNotEquivelentModel = newConsummingRoute("broker1", "exchange2", "queue1", "route1");
        _routeNotEquivelentModel = newConsummingRoute("broker1", "exchange1", "queue1", "route2");
        _queueNotEquivelentModel = newConsummingRoute("broker1", "exchange1", "queue2", "route1");
	}
	
    private ConsumingRoute newConsummingRoute(String host, String exchange, String queue, String routeKey){
    	ConsumingRoute route = new ConsumingRoute();
    	
    	route.brokers.add(new Broker(host, 0, "basic"));
    	route.exchange = new Exchange(exchange, "direct", true, true, true, null);
    	route.queue = new Queue(queue, true, true, true, true, null);
    	route.routingKeys.add(routeKey);
  	
        return route;
    }
    
	@Test
	public void Equals_should_return_false_if_queue_values_differ(){
		assertFalse(_referenceModel.equals(_queueNotEquivelentModel));
	}
	
	@Test
	public void Hashcodes_should_differ_if_queue_entries_differ(){
		assertNotEquals(_referenceModel.hashCode() , _queueNotEquivelentModel.hashCode());
	}
}
